addons/base_address_extended/views/base_address_extended.xml

```diff
@@ -1,90 +1,67 @@
-                    <field name="street" class="oe_read_only"/>
-                </div>
-                <field name="street_name" placeholder="Street Name..." attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}" class="oe_edit_only"/>
-                <div class="oe_edit_only o_row">
-                    <label for="street_number"/>
-                    <span> </span>
-                    <field name="street_number" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                    <label for="street_number2"/>
-                    <field name="street_number2" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                </div>
-            </xpath>
+     <record id="base.mx" model="res.country">
+         <field name="address_view_id" ref="address_street_extended_form" />
+     </record>
 
-        </field>
-    </record>
-
-    <record id="view_partner_address_structured_form" model="ir.ui.view">
-        <field name="name">view_partner_address_structured_form</field>
-        <field name="model">res.partner</field>
-        <field name="inherit_id" ref="base.view_partner_address_form"/>
-        <field name="arch" type="xml">
-            <xpath expr="//field[@name='street']" position="replace">
-                <field name="street" invisible="1"/>
-                <field name="street_name" placeholder="Street Name..." attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                <div class="o_row">
-                    <label for="street_number" class="oe_edit_only"/>
-                    <span> </span>
-                    <field name="street_number" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                    <label for="street_number2" class="oe_edit_only"/>
-                    <field name="street_number2" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                </div>
-            </xpath>
-        </field>
-    </record>
-
-    <record id="view_res_company_extended_form" model="ir.ui.view">
-        <field name="name">view_res_company_extended_form</field>
-        <field name="model">res.company</field>
-        <field name="inherit_id" ref="base.view_company_form"/>
-        <field name="arch" type="xml">
-            <xpath expr="//field[@name='street']" position="replace">
-                <field name="street" invisible="1"/>
-                <field name="street_name" placeholder="Street Name..."/>
-                <div class="o_row">
-                    <label for="street_number" class="oe_edit_only"/>
-                    <span> </span>
-                    <field name="street_number"/>
-                    <label for="street_number2" class="oe_edit_only"/>
-                    <field name="street_number2"/>
-                </div>
-            </xpath>
-        </field>
-    </record>
+     <record id="base.co" model="res.country">
```

Comment:
what about nl, de, hr? - they also had different address formats before
Can we not apply something similar for countries that require city_id - I think cn is one?

Reply:
nl seems only required for reports
cn_city depends on base_address_extended
check de, hr?

Reply:
Here's the analysis...

|l10n|city_id|enforce_cities|requires split street, number|depends base_address_extended|street_split|address form|
|----|-------|--------------|-----------------------------|-----------------------------|------------|------------|
|co| &check; | &check; |  | &check; | | extended|
|mx| &check;| &check; | &check; | &check; | | &check;|
|cn_city| &check; | ~&cross;~ &check; | | &check; | | ~?~ extended|
|cl||||&cross;|||
|pe|&check;|&check;||&check;||&check;|
|nl|||&check;||&check;||
|hr|||||||
|de||||||

@jco-odoo can you please confirm we can remove the base_address_extended dependecy from `l10n_cl`. Also, `l10n_cn_city` seems to define `city_id`'s, but does not modify the form - I have changed that.

Reply:
ok for l10n_cl (also we can remove the dependency on base_address_city).  For l10n_cn_city, it stays a weird module to me as the only thing it does, is provide cities for cn, but why would not we do it with one module for all city hungry countries of the world?  But we have it, so we could keep it and the only thing we would need to do, which is already the case is set enforce_cities on the China country to true.  

Because it might be a lot easier to apply the city_id to all the addresses (as was done before with the inherit of fields_view_get before, but we just apply it through the view here).  If you send from the Netherlands to China and you have l10n_cn_city installed then you can also immediately choose the city_id instead of typing it manually (I hope you have the zip as city names are in Chinese) and the address would not appear differently if you look at it from one company or the other.

Reply:
done